pypoptools/pypoptesting/framework/generate_util.py

```diff
@@ -0,0 +1,74 @@
+import itertools
+import random
+import time
+
+from .random_miner import RandomPopMiner
+
+
+def generate_endorsed(node, mock_miner, atvs, vtbs, alt_blocks, vbk_blocks, btc_blocks, seed, timeout=60):
+    m = RandomPopMiner(node, mock_miner, seed)
+
+    commands = chain_random(
+        [
+            itertools.repeat(m.mine_alt_block, times=alt_blocks),
+            itertools.repeat(m.mine_btc_block, times=btc_blocks),
+            iter(random.choices(
+                [lambda: m.mine_vbk_block(pop=True), lambda: m.mine_vbk_block(pop=False)],
+                weights=[atvs, vtbs],
+                k=vbk_blocks)),
+            iter(parentheses_sequence(m.submit_btc_tx, m.submit_vbk_pop_tx, n=vtbs)),
+            iter(parentheses_sequence(m.submit_vbk_tx, m.submit_alt_pop_data, n=atvs))
+        ],
+        sizes=[
+            alt_blocks,
+            btc_blocks,
+            vbk_blocks,
+            2 * vtbs,
+            2 * atvs
+        ]
+    )
+
+    start = time.time()
+    for command in commands:
+        if time.time() - start >= timeout:
+            break
+        command()
+
+
+def parentheses_sequence(open_object, close_object, n):
```

Comment:
please add either link to algo description, or a comment

Reply:
fixed